include/pybind11/pytypes.h

```diff
@@ -1171,11 +1171,22 @@ class weakref : public object {
 class slice : public object {
 public:
     PYBIND11_OBJECT_DEFAULT(slice, object, PySlice_Check)
-    slice(ssize_t start_, ssize_t stop_, ssize_t step_) {
-        int_ start(start_), stop(stop_), step(step_);
+
+    slice(handle start, handle stop, handle step) {
         m_ptr = PySlice_New(start.ptr(), stop.ptr(), step.ptr());
-        if (!m_ptr) pybind11_fail("Could not allocate slice object!");
+        if (!m_ptr)
+            pybind11_fail("Could not allocate slice object!");
     }
+
+#ifdef PYBIND11_HAS_OPTIONAL
```

Comment:
This is too early of a usage for `PYBIND11_HAS_OPTIONAL`.
It's defined in `stl.h`, but this depends on `pytest.h` indirectly via inclusion of `pybind11.h`.

Suggested fix: Consolidate STL usages iff any STL types are to be used in core API.

Reply:
Done in 5837ece4bfc